chainer/functions/pooling/unpooling_2d.py

```diff
@@ -55,7 +56,22 @@ def forward(self, x):
                                 self.outh, self.outw)
         return y,
 
-    def backward(self, x, gy):
+    def backward(self, indexes, grad_outputs):
+        return Unpooling2DGrad(self).apply(grad_outputs)
+
+
+class Unpooling2DGrad(function_node.FunctionNode):
```

Comment:
Is it correct that this function has the gradient always begin zero?

Reply:
Sorry, I should have implemented the backward for this class. So, I added it.